app/views/main/_past_contests.html.erb

```diff
@@ -1,32 +1,48 @@
 <h2>Предишни състезания</h2>
 <br />
-<% unless @past_contests.blank? %>
-  <%= will_paginate @past_contests, :param_name => :past_contests_page %>
-  <table>
-    <tr>
-      <th>Име</th>
-      <th>Начало</th>
-      <th>Край</th>
-      <th>Продължителност</th>
-      <th>Резултати</th>
-    </tr>
-    <% for contest in @past_contests do %>
-      <tr>
-        <td><%= contest.name %></td>
-        <td><%= contest.start_time.to_s :long %></td>
-        <td><%= contest.end_time.to_s :long %></td>
-        <td><%= duration_in_words contest.duration.minutes %></td>
-        <td>
-          <% if contest.results_visible? or current_user.andand.admin? %>
-            <%= link_to "Резултати", :action => :results, :contest_id => contest.id %>
-          <% else %>
-            Очаквайте скоро
-          <% end %>
-        </td>
-      </tr>
+
+<% unless @contest_groups.blank? %>
+    <table>
+        <tr>
+    <% @contest_groups.each do |g| %>
+            <td onclick="show_contest_group('<%= g.id %>')"><%= g.name %></td>
     <% end %>
-  </table>
-  <%= will_paginate @past_contests, :param_name => :past_contests_page %>
-<% else %>
-  Няма предишни състезания.
-<% end %>
\ No newline at end of file
+        </tr>
+    </table>
+<% end %>
+<br />
+
+<% @sort_contests.each_with_index do |c, index| %>
+    <div id="groupid_<%= c[0]%>" style="<%= (c[0].to_i != 1 ? "display:none;" : '') %>" >
```

Comment:
There is some complex logic in the template here. It will be much better to extract it into a helper, like this:

```erb
<div id="groupid_<%= c[0]%>" style="<%= contest_group_style(c[0]) %>" >
```

And then define the helper in the `main_helper.rb` file like this:

```ruby
def contest_group_style(contest_group_id)
  return unless contest_group_id.to_i != 1
  "display: none;"
end
```